In [1]:
import pandas as pd

In [2]:
ds = pd.read_csv("Placement.csv")

In [3]:
#ds.info()

In [4]:
from Univariate import Univariate
Qual,Quan = Univariate.QualQuan(ds)

In [5]:
Qual

['gender',
 'ssc_b',
 'hsc_b',
 'hsc_s',
 'degree_t',
 'workex',
 'specialisation',
 'status']

In [6]:
Quan

['sl_no', 'ssc_p', 'hsc_p', 'degree_p', 'etest_p', 'mba_p', 'salary']

# Central Tendency

In [32]:
ds.describe()

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
count,215.000000,215.000000,215.000000,215.000000,215.000000,215.000000,148.000000
mean,108.000000,67.303395,66.333163,66.370186,72.100558,62.278186,288655.405405
std,62.209324,10.827205,10.897509,7.358743,13.275956,5.833385,93457.452420
min,1.000000,40.890000,37.000000,50.000000,50.000000,51.210000,200000.000000
25%,54.500000,60.600000,60.900000,61.000000,60.000000,57.945000,240000.000000
50%,108.000000,67.000000,65.000000,66.000000,71.000000,62.000000,265000.000000
75%,161.500000,75.700000,73.000000,72.000000,83.500000,66.255000,300000.000000
max,215.000000,89.400000,97.700000,91.000000,98.000000,77.890000,940000.000000


In [7]:
ds["ssc_p"].mean() # Average Value

np.float64(67.30339534883721)

In [8]:
ds["ssc_p"].median() # Center Value

np.float64(67.0)

In [9]:
ds["ssc_p"].mode() # Repeated value

0    62.0
Name: ssc_p, dtype: float64

In [10]:
ds["ssc_p"].mode() [0]# Repeated value

np.float64(62.0)

In [11]:
descriptive=pd.DataFrame(index=["Mean","Median","Mode"],columns=Quan)

In [12]:
descriptive

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
Mean,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Median,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Mode,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [13]:
descriptive.loc["ssc_p","Mode"]=0

In [14]:
descriptive=pd.DataFrame(index=["Mean","Median","Mode"],columns=Quan)
for columnName in Quan:
    descriptive.loc["Mean",columnName]=ds[columnName].mean()
    descriptive.loc["Median",columnName]=ds[columnName].median()
    descriptive.loc["Mode",columnName]=ds[columnName].mode()[0]

In [15]:
descriptive

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
Mean,108.0,67.303395,66.333163,66.370186,72.100558,62.278186,288655.405405
Median,108.0,67.0,65.0,66.0,71.0,62.0,265000.0
Mode,1,62.0,63.0,65.0,60.0,56.7,300000.0


# Percentile

In [16]:
import numpy as np

In [19]:
np.percentile(ds["ssc_p"],25)

np.float64(60.599999999999994)

In [36]:
descriptive=pd.DataFrame(index=["Mean","Median","Mode","Q1:25%","Q2:50%","Q3:75%","99%","Q4:100"],columns=Quan)
for columnName in Quan:
    descriptive.loc["Mean",columnName]=ds[columnName].mean()
    descriptive.loc["Median",columnName]=ds[columnName].median()
    descriptive.loc["Mode",columnName]=ds[columnName].mode()[0]
    descriptive.loc["Q1:25%",columnName]=ds.describe()[columnName]["25%"]
    descriptive.loc["Q2:50%",columnName]=ds.describe()[columnName]["50%"]
    descriptive.loc["Q3:75%",columnName]=ds.describe()[columnName]["75%"]
    descriptive.loc["99%",columnName]=np.percentile(ds[columnName],99)
    descriptive.loc["Q4:100",columnName]=ds.describe()[columnName]["max"]

In [39]:
descriptive

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
Mean,108.0,67.303395,66.333163,66.370186,72.100558,62.278186,288655.405405
Median,108.0,67.0,65.0,66.0,71.0,62.0,265000.0
Mode,1,62.0,63.0,65.0,60.0,56.7,300000.0
Q1:25%,54.5,60.6,60.9,61.0,60.0,57.945,240000.0
Q2:50%,108.0,67.0,65.0,66.0,71.0,62.0,265000.0
Q3:75%,161.5,75.7,73.0,72.0,83.5,66.255,300000.0
99%,212.86,87.0,91.86,83.86,97.0,76.1142,NaN
Q4:100,215.0,89.4,97.7,91.0,98.0,77.89,940000.0


# hsc_p, Difference between 25th and 50th is 4%, 50th to 75th is 8%, 75th to 100 is 14.7%, 99th to 100 is 6%
# degree_p, Difference between 25th and 50th is 5%, 50th to 75th is 6%, 75th to 100 is 19%, 99th to 100 is 7%
# etest_p, Difference between 25th and 50th is 11%, 50th to 75th is 12%, 75th to 100 is 15%, 99th to 100 is 1%
# mba_p, Difference between 25th and 50th is 4%, 50th to 75th is 4%, 75th to 100 is 11%, 99th to 100 is 1%